app/webpacker/components/CompetitionResultSubmission/UploadResultsJson.jsx

```diff
@@ -0,0 +1,42 @@
+import { useMutation } from '@tanstack/react-query';
+import React, { useState } from 'react';
+import { Form } from 'semantic-ui-react';
+import uploadResultsJson from './api/uploadResultsJson';
+import useCheckboxState from '../../lib/hooks/useCheckboxState';
+import Errored from '../Requests/Errored';
+
+export default function UploadResultsJson({ competitionId, isWrtViewing }) {
+  const [resultFile, setResultFile] = useState();
+  const [markResultSubmitted, setMarkResultSubmitted] = useCheckboxState(isWrtViewing);
+
+  const { mutate: uploadResultsJsonMutate, error, isError } = useMutation({
+    mutationFn: () => uploadResultsJson({ competitionId, resultFile, markResultSubmitted }),
+    onSuccess: () => {
+      window.location.reload();
+    },
+  });
+
+  if (isError) return <Errored error={error} />;
+
+  return (
+    <Form>
+      <Form.Input
+        type="file"
+        onChange={(event) => setResultFile(event.target.files[0])}
+      />
+      {isWrtViewing && (
+        <Form.Checkbox
+          checked={markResultSubmitted}
+          onChange={setMarkResultSubmitted}
+          label="If results are not marked as submitted, mark it as submitted (this is only visible to WRT)"
+        />
+      )}
+      <Form.Button
+        disabled={!resultFile}
+        onClick={uploadResultsJsonMutate}
```

Comment:
You should make this button `type="submit"` and move the mutation to the `onSubmit` of the form. This is because `onSubmit` will run browser-sided form validations for you (like checking `required` fields for example). 
May not be too useful here because your form isn't exactly complex, but still a good rule of thumb in general to use `onSubmit` whenever you can